linera-base/src/data_types.rs

```diff
@@ -171,6 +170,19 @@ pub struct Resources {
     // TODO(#1533): Allow declaring calls to other applications instead of having to count them here.
 }
 
+/// The kind of outgoing message being sent.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub enum MessageKind {
+    /// The message can be skipped or rejected. No receipt is requested.
+    Simple,
+    /// The message cannot be skipped nor rejected. No receipt is requested.
+    /// This only concerns certain system messages that cannot fail.
+    Protected,
```

Comment:
You won't be able to share `MessageKind` with Wasm because user apps are not allowed to send protected messages.